include/ImmediateContext.hpp

```diff
@@ -12,6 +12,112 @@ struct TranslationLayerCallbacks
     std::function<void()> m_pfnPostSubmit;
 };
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// A pool of objects that are recycled on specific fence values
+// with a maximum depth before blocking on RetrieveFromPool
+// This class assumes single threaded caller
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template<typename TResourceType>
+class CBoundedFencePool

```

Comment:
How bad would it be to have this inherit from `CFencePool`? Or to replace `CFencePool` with this instead?